react-d3-chart/src/components/ModeSelectionContainer/index.tsx

```diff
@@ -23,9 +37,33 @@ const ModeSelectionContainer: React.FC<Props> = ({
   width,
   mode,
   selection,
+  onConfirmSelection,
+  setRangeSelectionState,
+  setRangeSelections,
+  rangeSelections,
 }) => {
+
+  const addSelection = () => {
+    onConfirmSelection(selection);
+    setRangeSelectionState({
+      selection: [0, 0],
+      eventSource: 'reset'
```

Comment:
Potential for using enum here?

Reply:
this field is not really used - i will mark it as optional.